src/Symfony/Component/HttpKernel/Kernel.php

```diff
@@ -672,10 +672,16 @@ protected function dumpContainer(ConfigCache $cache, ContainerBuilder $container
      *
      * @param ContainerInterface $container The service container
      *
+     * @throws \InvalidArgumentException When the container is not an instance of ContainerBuilder
+     *
      * @return DelegatingLoader The loader
      */
     protected function getContainerLoader(ContainerInterface $container)
     {
+        if (!$container instanceof ContainerBuilder) {
+            throw new \InvalidArgumentException(sprintf('The container must be an instance of Symfony\Component\DependencyInjection\ContainerBuilder, got %s.', get_class($container)));
```

Comment:
It would be "won't" fix even in 4.0 because we have a policy of:
1. issuing a deprecation notice in the current version
2. allow people to fix their in the same version where the notice appeared
3. *then* make the change. If the notice  can't 

Since 2. is not possible here, neither is 3. This ensures a continuous upgrade path.